pkg/prometheus/prometheus.go

```diff
@@ -0,0 +1,52 @@
+package prometheus
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const prometheusDefaultPort = "2112"
+
+var (
+	ApiCallsFailure = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "topology_updater_api_call_failures_total",
+		Help: "The total number of api calls that failed by the updater",
+	})
+
+	OperationDelay = promauto.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "topology_updater_hw_topology_update_operation_measurement",
+			Help: "Represent the latency of the update operation",
```

Comment:
ok, the core concept is there, we will need to add labels to distinguish between the operations, and we will probably need one or two more metrics